projects/commudle-admin/src/app/services/stepper.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@angular/core';
+import { NbDialogRef, NbDialogService } from '@nebular/theme';
+import { ICurrentUser } from 'projects/shared-models/current_user.model';
+import { LibAuthwatchService } from 'projects/shared-services/lib-authwatch.service';
+import { BehaviorSubject } from 'rxjs';
+import { StepperComponent } from 'projects/commudle-admin/src/app/components/stepper/stepper.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StepperService {
+  // user basic info = 70% , total fields = 7, each weight = 10
+  // social links = 30%, total fields = 10, each weight = 3
+  private profileWeights: Object = {
+    username: 10,
+    avatar: 10,
+    name: 10,
+    designation: 10,
+    about_me: 10,
+    location: 10,
+    gender: 10,
+    skills: 20,
+    socialLinks: 10,
+  };
+
+  private socialLinks = [
```

Comment:
do we need a separate array for the social links? can we merge it into the object above?

Reply:
we need to keep it separate.